fastlane_core/lib/fastlane_core/helper.rb

```diff
@@ -299,6 +299,22 @@ def self.zip_directory(path, output_path, contents_only: false, overwrite: false
       Helper.backticks(command, print: print)
     end
 
+    # Executes the provided block after adjusting the ENV to have the
+    # provided keys and values set as defined in hash. After the block
+    # completes, restores the ENV to its previous state.
+    def self.with_env_values(hash, &block)
```

Comment:
Nice!

Reply:
Oh, I fully stole that from the screengrab `spec_helper`:
https://github.com/fastlane/fastlane/blob/89da896d167f8c7ec33c05601ffe37a182e788d1/screengrab/spec/spec_helper.rb#L1-L23 
So all credit to @mfurtak 🎁